File: Go/003-Largest_prime_factor.go
package main

import "fmt"

func prime(n int) bool {
	for i:=2; i<n; i++ {
		if n % i == 0 {
			return false
		}
	}
	return true
}

func main() {
	number := 600851475143
	for i:=1; i <= number; i++ {
		if prime(i) && number % i == 0{
			number /= i
			fmt.Printf("%d\n", i)
		}
	}
}

File: Python/003_Largest_prime_factor.py
number = 600851475143


def isPrime(n):
    for i in range(2, n):
        if n % i == 0 and i != n:
            return False
    return True


def LargestPrimeFactor(n):
    _n, lpf = n, 0
    for i in range(2, n):
        if isPrime(i) and n % i == 0:
            _n, lpf = _n/i, i
            print(i, end=', ')
        if _n == 1:
            break
    print(lpf)
    return


LargestPrimeFactor(number)
